sympy/polys/numberfields.py

```diff
@@ -518,7 +518,8 @@ def _minpoly_compose(ex, x, dom):
         _, factors = factor_list(x**2 + 1, x, domain=dom)
         return x**2 + 1 if len(factors) == 1 else x - I
     if ex is GoldenRatio:
-        return x**2 - x - 1
+        _, factors = factor_list(x**2 - x - 1, x, domain=dom)
+        return x**2 - x - 1 if len(factors) == 1 else x - GoldenRatio
```

Comment:
The result should probably not contain `GoldenRatio` (danger of circularity). Instead, one should choose one of the factors using `_choose_factor` with argument `(1 + sqrt(5))/2`.